spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkCatalogEventProcessor.scala

```diff
@@ -33,105 +33,107 @@ class SparkCatalogEventProcessor(
-      case CreateTableEvent(db, table) =>
-        val tableDefinition = SparkUtils.getExternalCatalog().getTable(db, table)
-        val tableEntity = tableToEntity(tableDefinition)
-        atlasClient.createEntitiesWithDependencies(tableEntity)
-        logDebug(s"Created table entity $table without columns")
-
-      case DropTablePreEvent(_, _) => // No-op
-
-      case DropTableEvent(db, table) =>
-        logDebug(s"Can't handle drop table event since we don't have context information for " +
-          s"table $table in db $db. Can't delete table entity and corresponding entities.")
-
-      case RenameTableEvent(db, name, newName) =>
-        val tableDefinition = SparkUtils.getExternalCatalog().getTable(db, newName)
-        val isHiveTbl = isHiveTable(tableDefinition)
-
-        // Update storageFormat's unique attribute
-        val sdEntity = new AtlasEntity(storageFormatType(isHiveTbl))
-        sdEntity.setAttribute(org.apache.atlas.AtlasClient.REFERENCEABLE_ATTRIBUTE_NAME,
-          storageFormatUniqueAttribute(db, newName, isHiveTbl))
-        atlasClient.updateEntityWithUniqueAttr(
-          storageFormatType(isHiveTbl),
-          storageFormatUniqueAttribute(db, name, isHiveTbl),
-          sdEntity)
-
-        // Update Table name and Table's unique attribute
-        val tableEntity = new AtlasEntity(tableType(isHiveTbl))
-        tableEntity.setAttribute(org.apache.atlas.AtlasClient.REFERENCEABLE_ATTRIBUTE_NAME,
-          tableUniqueAttribute(db, newName, isHiveTbl))
-        tableEntity.setAttribute("name", newName)
-        atlasClient.updateEntityWithUniqueAttr(
-          tableType(isHiveTbl),
-          tableUniqueAttribute(db, name, isHiveTbl),
-          tableEntity)
-
-        logDebug(s"Rename table entity $name to $newName")
-
-      case AlterDatabaseEvent(db) =>
-        val dbDefinition = SparkUtils.getExternalCatalog().getDatabase(db)
-        val dbEntity = dbToEntity(dbDefinition)
-        atlasClient.createEntitiesWithDependencies(dbEntity)
-        logDebug(s"Updated DB properties")
-
-      case AlterTableEvent(db, table, kind) =>
-        val tableDefinition = SparkUtils.getExternalCatalog().getTable(db, table)
-        kind match {
-          case "table" =>
-            val tableEntity = tableToEntityForAlterTable(tableDefinition)
-            atlasClient.createEntitiesWithDependencies(tableEntity)
-            logDebug(s"Updated table entity $table without columns")
-
-          case "dataSchema" =>
-            // We don't mind updating column
-            logDebug("Detected updating of table schema but ignored: " +
-              "column update will not be tracked here")
-
-          case "stats" =>
-            logDebug(s"Stats update will not be tracked here")
-
-          case _ =>
-          // No op.
-        }
-
-      case f =>
-        logDebug(s"Drop unknown event $f")
+    if (SparkUtils.usingRemoteMetastoreService()) {
```

Comment:
SAC will not handle any DDL events when remote HMS is used: Hive hook will take care of it.

Reply:
Do nothing for remote HMS only applies to external catalog events which ships DDL. For other processors they should handle the remote HMS case as well.

Reply:
Will elaborate the code comment. Nice suggestion.

Reply:
What I meant was in `SparkAtlasEventTracker` you could have something like,
```java 
val catalogEventTracker = if (remoteMetaStore) new NoopProcessor() 
else new SparkCatalogEventProcessor(atlasClient, atlasClientConf)
```
The proposed approach is also ok so leave it to you.

Reply:
Ah OK now I see the point. I'd still want to block the case where remoteMetastore is being used in SparkCatalogEventProcessor, but introducing `NoopProcessor()` and adding `require` would simply work. Nice suggestion.

Reply:
Looks like there're lots of cases where SparkCatalogEventProcessor should be referred - more likely, `AtlasEntityUtils` mixed-in should be needed. We can just allow to use `return` in this case so that we just ensure here we change couple of lines instead.